### PR TITLE
[SPARK-7604] [MLlib] Python API for PCA and PCAModel

### PR DESCRIPTION
**Pull request by yanboliang**
Python API for PCA and PCAModel

### CODE REVIEW

File: python/pyspark/mllib/feature.py [hunk 1]
```diff
@@ -254,6 +254,49 @@ def fit(self, data):
...
+    Model fitted by [[PCA]] that can project vectors to a low-dimensional space using PCA.
+    """
+    def transform(self, vector):
+        """
+        Applies transformation on a vector.
+
+        :param vector: Vector or RDD of Vector to be transformed.
+        :return: transformed vector.
+        """
+        return JavaVectorTransformer.transform(self, vector)
```
Comment (davies) [inline python/pyspark/mllib/feature.py]: This is the default behavior, so you could not have `transform()` in `PCAModel`.

**Comment by yanboliang** [comment]
@mengxr @jkbradley

**Comment by yanboliang** [comment]
@jkbradley @mengxr

**Comment by jkbradley** [comment]
LGTM  sorry for the delay!
I'll rerun tests since it's been a while, before merging

**Comment by jkbradley** [comment]
test this please

**Comment by jkbradley** [comment]
Merging with master and branch-1.4

Thanks!

**Pull request closed without merging**